### DISCUSSION TITLE
.3mf file association, SuperSlicer does not launch when attempting to open .3mf files

### DISCUSSION THREAD

**Discussion by ids444** [supermerill/SuperSlicer#1818] [Q&A]
Windows 10 and 11, SuperSlicer 2.3.57.3 & .4

.3mf files are associated with SuperSlicer
double clicking on file or right click ans selecting open, does not launch SuperSlicer. Works as expected with .stl

PrusaSlicer works as expected.

**Comment by supermerill** [comment]
It works well for me.
Can you describe exactly how it fails?

**Reply by ids444** [reply]
I double click on a 3mf file and nothing happens

Change the open with to Prusa and it opens in Prusa, change it back to superslicer and cursor flickers to busy then back to arrow and nothing launches. 

I first installed SuperSlicer into C:\Program Files\SuperSlicer_2.3.57.3_win64_211022 and setup the open with at that time. Then I renamed the folder to just SuperSlicer (not sure if I ever opened a .3mf file before renaming) but I renamed the folder back to SuperSlicer_2.3.57.3_win64_211022 still didn't work.

Searched the registry for references to SuperSlicer_2.3.57.3_win64_211022 and deleted them all. But the reference to Superslicer looks good to me

```
[HKEY_USERS\S-1-5-21-446955144-2225154037-2407054437-1002_Classes\SuperSlicer.1]
@="SuperSlicer"

[HKEY_USERS\S-1-5-21-446955144-2225154037-2407054437-1002_Classes\SuperSlicer.1\Shell]

[HKEY_USERS\S-1-5-21-446955144-2225154037-2407054437-1002_Classes\SuperSlicer.1\Shell\Open]

[HKEY_USERS\S-1-5-21-446955144-2225154037-2407054437-1002_Classes\SuperSlicer.1\Shell\Open\Command]
@="\"C:\\Program Files\\SuperSlicer\\superslicer.exe\" \"%1\""
```

https://user-images.githubusercontent.com/10389439/140650490-499a93fa-611a-440f-900d-8fa70c970984.mp4

**Reply by supermerill** [reply]
I added the superslicer.exe via "open with..." -> search another app
and then when I open the .3mf and select superslicer, it works like prusaslicer did in your video.
the command in my regedit is `"C:\Users\MYNAME\3D Objects\Susie\SuperSlicer_2.3.57.3_win64_211022\superslicer.exe" "%1"`

if you drag & drop a .3mf on the superslicer.exe, it should do the same things, does it works?
or launching it via the cmd with the .3mf as an argument?
Does loading the .3mf in the app via ctrl+O works?

**Comment by ids444** [comment]
> if you drag & drop a .3mf on the superslicer.exe, it should do the same things, does it works?

Yes

or launching it via the cmd with the .3mf as an argument?

No

> Does loading the .3mf in the app via ctrl+O works?

Yes

Just tried with different 3mf files (duh), it is only doing it for 3mf's from PrusaPrinters.org, https://www.prusaprinters.org/prints/23923-lack-enclosure-v2-modified-to-print-on-the-mini, none of these open in SuperSlicer. Also this https://www.prusaprinters.org/prints/8018-ikea-dry-box-rollers-4-spools-of-prusament has one 3mf in the project and it too won't open with Superslicer.

While ones I have created myself in PrusaSlicer, Fusion 360 or SuperSlicer open fine. And just tried a file that won't open, loaded it into Superslicer with file open then re-saved it and it opens, attaching both.

[enclosure-part-10-super.zip](https://github.com/supermerill/SuperSlicer/files/7492794/enclosure-part-10-super.zip)

**Reply by supermerill** [reply]
Found the problem.

Next time, open an issue, as i can't close a discussion, only delete it.
